### PR TITLE
Made the change so the you only see friends in the usertable

### PR DESCRIPTION
**Pull request by ronakJain21**

### CODE REVIEW

File: client/src/components/Direct-Messaging/DirectMessaging.jsx [hunk 1]
```diff
@@ -18,13 +18,12 @@ const DirectMessaging = () => {
     useEffect(() => {
         const fetchUsersAndMessages = async () => {
             try {
-                const usersResponse = await fetch('http://localhost:8800/users');
-                const usersData = await usersResponse.json();
-                if (usersResponse.ok) {
-                    const filteredUsers = usersData.filter(user => user.id !== currentUser.id);
-                    setUsers(filteredUsers);
+                const friendsResponse = await fetch(`http://localhost:8800/friends-list/${currentUser.id}`);
+                const friendsData = await friendsResponse.json();
```
Comment (aisaaxs) [inline client/src/components/Direct-Messaging/DirectMessaging.jsx:22]: Usually, if you type friendsResponse.data, it returns a json object, so you don't have to use the json() function for it or create another variable with the keyword await as the data has already been fetched under the friendsResponse constant. Regardless, as long as it works, it's all good, but if you'd like to be consistent in terms of the approach, consider making the change.

File: client/src/components/Direct-Messaging/DirectMessaging.jsx [hunk 2]
```diff
@@ -18,13 +18,12 @@ const DirectMessaging = () => {
...
+                if (friendsResponse.ok) {
```
Comment (aisaaxs) [inline client/src/components/Direct-Messaging/DirectMessaging.jsx:23]: You don't really have to use ".ok". Just using "if (friendsResponse)" would do the trick. But again, if it works, it's all good.

File: client/src/components/Direct-Messaging/DirectMessaging.jsx [hunk 3]
```diff
@@ -149,17 +148,19 @@ const DirectMessaging = () => {
...
-                    <div
-                        key={user.id}
-                        onClick={() => selectUser(user.id)}
-                        className={`user-item ${user.id === receiverId ? 'selected' : ''}`}
-                    >
-                        {user.name}
-                    </div>
-                ))}
-            </aside>
+        <aside className="user-sidebar">
```
Comment (aisaaxs) [inline client/src/components/Direct-Messaging/DirectMessaging.jsx:151]: An aside tag usually refers to a portion of a document whose content is only indirectly related to the document's main content, but this portion is directly related to the component as it's responsible for showing the user's existing DMs, so consider using a <div></div> instead.

This is just a suggestion and you're free to use any appropriate approach as you see fit.

File: client/src/__test__/DirectMessaging.test.js [hunk 4]
```diff
@@ -5,15 +5,14 @@ import DirectMessaging from '../components/Direct-Messaging/DirectMessaging';
 
 beforeEach(() => {
   jest.clearAllMocks();
-
+  
   global.fetch = jest.fn((url) =>
     Promise.resolve({
       json: () => {
-        if (url.endsWith('/users')) {
+        if (url.endsWith('/friends-list/123')) {
```
Comment (aisaaxs) [inline client/src/__test__/DirectMessaging.test.js:12]: I'm not sure if you've implemented this already, but please make sure that a user can't access the component or its features by simply entering the url into their browser as that can cause a ton of complications. Therefore, please consider limiting access to the component to a logged in user only.

File: client/src/__test__/DirectMessaging.test.js [hunk 5]
```diff
@@ -42,16 +41,16 @@ describe('DirectMessaging Component', () => {
...
   });
 
   test('updates receiverId when clicking on a user', async () => {
     render(<DirectMessaging />);
-    fireEvent.click(await screen.findByText('User One'));
-    expect(Storage.prototype.setItem).toHaveBeenCalledWith('receiverId', 'user1');
+    fireEvent.click(await screen.findByText('User Two'));
+    expect(Storage.prototype.setItem).toHaveBeenCalledWith('receiverId', 'user2');
   });
 });
```
Comment (aisaaxs) [inline client/src/__test__/DirectMessaging.test.js:56]: As long as it works, it's all good, but endpoints appear a little loose in terms of a user gaining access to the features of the component simply by typing in the url in address bar of their browser. So, try revising the component or its endpoints accordingly. Another thing to note is that, test out ever little detail of your component. If it's unit testing, you have to test the UI of the component as well as the backend/functionality. And if it's integration testing, you have to test how well the component is working with other components.

File: client/src/__test__/DirectMessaging.test.js (hunk 4 above)
Comment (ronakJain21) [reply client/src/__test__/DirectMessaging.test.js:12]: I have already done this

File: client/src/components/Direct-Messaging/DirectMessaging.jsx (hunk 1 above)
Comment (ronakJain21) [reply client/src/components/Direct-Messaging/DirectMessaging.jsx:31]: I've followed your advice to streamline our code. Now, I'm using await friendsResponse.json() directly for fetching and parsing data, which keeps our approach consistent and clean. Thanks for pointing that out!

**Review by ronakJain21 — COMMENTED** [review]
I made the changes you asked fo

**Review by SoumilChhabra — APPROVED** [review]
Checked and test. Works perfectly!!

**Pull request merged**